src/core/CoreModuleDirector.ts

```diff
@@ -291,40 +330,58 @@ export class CoreModuleDirector {
   }
 
   public getSubscriptionOfTypeWithToken(
-    type: ModelName,
+    type: SubscriptionType | ModelName.Subscriptions,
     token: string,
   ): OSModel<SupportedSubscription> | undefined {
     logMethodCall('CoreModuleDirector.getSubscriptionOfTypeWithToken', {
       type,
       token,
     });
+
+    let subscriptions: Record<string, OSModel<SupportedSubscription>>;
+
     switch (type) {
-      case ModelName.EmailSubscriptions: {
-        const emailSubscriptions = this.getEmailSubscriptionModels();
-        return Object.values(emailSubscriptions).find(
-          (subscription) => subscription.data.token === token,
-        );
-      }
-      case ModelName.SmsSubscriptions: {
-        const smsSubscriptions = this.getSmsSubscriptionModels();
-        return Object.values(smsSubscriptions).find(
-          (subscription) => subscription.data.token === token,
-        );
-      }
-      case ModelName.PushSubscriptions: {
-        const pushSubscriptions = this.getAllPushSubscriptionModels();
-        return Object.values(pushSubscriptions).find(
-          (subscription) => subscription.data.token === token,
-        );
-      }
+      case SubscriptionType.Email:
+        subscriptions = this.getEmailSubscriptionModels();
+        break;
+      case SubscriptionType.SMS:
+        subscriptions = this.getSmsSubscriptionModels();
+        break;
+      case ModelName.Subscriptions:
+        subscriptions = this.getAllPushSubscriptionModels();
+        break;
```

Comment:
This seems like a bad assumption (or atleast a bad name), `ModelName.Subscriptions` sounds like all subscriptions (including SMS and Email), but it getting a list of push subceptions only.

Reply:
I've created a new `SubscriptionChannel` enum: https://github.com/OneSignal/OneSignal-Website-SDK/pull/1179/commits/f260cab3ef4ee4e815e3eb569aa3101162be5cf3